src/data/list/nodup.lean

```diff
@@ -352,6 +352,18 @@ lemma nodup.pairwise_of_set_pairwise {l : list α} {r : α → α → Prop}
   (hl : l.nodup) (h : {x | x ∈ l}.pairwise r) : l.pairwise r :=
 hl.pairwise_of_forall_ne h
 
+@[simp] lemma nodup.pairwise_coe [is_symm α r] (hl : l.nodup) :
+  {a | a ∈ l}.pairwise r ↔ l.pairwise r :=
```

Comment:
Is the name better as `nodup.pairwise_coe_iff`? Also we don't actually have a coercion from `list` to `set` now.

Reply:
Hmm, I now wonder why [finset.set.has_coe_t](https://leanprover-community.github.io/mathlib_docs/data/finset/basic.html#finset.set.has_coe_t) is not via a `set_like` instance. (Of course `list` and `multiset` can't have `set_like` instances since taking membership predicate isn't injective.)

Reply:
I prefer the name as is to match the general convention from `list`/`multiset`/`finset` of omitting the `_iff` quite eagerly. However I agree that the `coe` bit is a bit bad because there won't ever be a coercion, as all coercions are meant to be injective.

Reply:
Maybe we can just fix that in a later PR? This is very tangential to the current.

Reply:
> all coercions are meant to be injective.

I don't think that's a universal rule? [int.cast_coe](https://leanprover-community.github.io/mathlib_docs/data/int/cast/defs.html#int.cast_coe) certainly aren't always injective, and #16567 adds has_lift_t for algebras which also uses `coe` but isn't always surjective.

Reply:
Hmm... true, but I don't know of a non-injective coercion that's explicitly between concrete types, as is the case here.

Reply:
> Maybe we can just fix that in a later PR? This is very tangential to the current.

I think so!

Reply:
Actually we have [multiset.has_coe](https://leanprover-community.github.io/mathlib_docs/data/multiset/basic.html#multiset.has_coe) from `list` to `multiset` which is a quotient map, so it's surjective and seldom injective. There are many other examples of quotient maps as coercions in mathlib as well:
[quotient_group.has_quotient.quotient.has_coe_t](https://leanprover-community.github.io/mathlib_docs/group_theory/coset.html#quotient_group.has_quotient.quotient.has_coe_t)
[quiver.weakly_connected_component.has_coe_t](https://leanprover-community.github.io/mathlib_docs/combinatorics/quiver/connected_component.html#quiver.weakly_connected_component.has_coe_t)
[add_circle.has_coe_t](https://leanprover-community.github.io/mathlib_docs/topology/instances/add_circle.html#add_circle.has_coe_t)

So I don't see why we shouldn't make `multiset.to_finset` a coercion and then by transitivity we get coercion from `list` all the way down to `set`.

Reply:
Oh, I think the problem is that `a ∈ m.to_finset` isn't be defeq to `a ∈ m` for `m : multiset α`. Maybe that's why `multiset.to_finset` isn't a coercion. But we may still have coercions from `multiset` and `list` to `set`, even though they're neither injective nor surjective ...